kolibri/core/auth/models.py

```diff
@@ -137,6 +138,23 @@ def __str__(self):
         else:
             return "FacilityDataset (no associated Facility)"
 
+    def save(self, *args, **kwargs):
+        try:
```

Comment:
I think better to just let this specific error be raised, so that we can distinguish it from an `IntegrityError` which will generally indicate some sort of database level constraint violation, which is not the case here.

Reply:
Got it. I have made this change.

Reply:
Can remove the extra 'try/except' clause here, and just call `self.ensure_compatibility()` and let the error raise from there - no need to reraise it.